### PR TITLE
Use vst in cluster

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

enable VST in the cluster

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)

### Testing & Verification

This change is already covered by existing tests, such as shell_client

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8430/

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8430/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8449/

File: arangod/Cluster/ClusterMethods.cpp [hunk 1]
```diff
@@ -1168,10 +1168,11 @@ Result selectivityEstimatesOnCoordinator(ClusterFeature& feature, std::string co
     if (tid != 0) {
       headers.try_emplace(StaticStrings::TransactionId, std::to_string(tid));
     }
-
+    
+    reqOpts.param("collection", StringUtils::urlEncode(p.first));
```
Comment (jsteemann) [inline arangod/Cluster/ClusterMethods.cpp]: won't this double-urlencode the parameter?

File: arangod/GeneralServer/VstCommTask.cpp [hunk 2]
```diff
@@ -244,7 +244,7 @@ bool VstCommTask<T>::processMessage(velocypack::Buffer<uint8_t> buffer,
       RequestStatistics::SET_SUPERUSER(stat);
     }
 
-    LOG_TOPIC("92fd6", DEBUG, Logger::REQUESTS)
+    LOG_TOPIC("92fd6", INFO, Logger::REQUESTS)
```
Comment (jsteemann) [inline arangod/GeneralServer/VstCommTask.cpp]: can't we keep this at DEBUG log level as before?

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 3]
```diff
@@ -391,6 +391,8 @@ Result DatabaseInitialSyncer::parseCollectionDump(transaction::Methods& trx,
   std::string const& cType =
       response->getHeaderField(StaticStrings::ContentTypeHeader, found);
   if (found && (cType == StaticStrings::MimeTypeVPack)) {
+    LOG_TOPIC("b9f4d", INFO, Logger::REPLICATION) << "using vpack for chunk contents";
```
Comment (jsteemann) [inline arangod/Replication/DatabaseInitialSyncer.cpp]:
```suggestion
    LOG_TOPIC("b9f4d", DEBUG, Logger::REPLICATION) << "using vpack for chunk contents";
```

File: arangod/Replication/DatabaseInitialSyncer.cpp [hunk 4]
```diff
@@ -427,6 +429,7 @@ Result DatabaseInitialSyncer::parseCollectionDump(transaction::Methods& trx,
   } else {
     // buffer must end with a NUL byte
     TRI_ASSERT(*end == '\0');
+    LOG_TOPIC("bad5d", INFO, Logger::REPLICATION) << "using json for chunk contents";
```
Comment (jsteemann) [inline arangod/Replication/DatabaseInitialSyncer.cpp]:
```suggestion
    LOG_TOPIC("bad5d", DEBUG, Logger::REPLICATION) << "using json for chunk contents";
```

File: arangod/RestHandler/RestReplicationHandler.cpp [hunk 5]
```diff
@@ -1434,14 +1434,17 @@ Result RestReplicationHandler::parseBatch(std::string const& collectionName,
 
   allMarkers.clear();
 
-  if (_request->transportType() != Endpoint::TransportType::HTTP) {
+  // simon: original VST was not allowed here, but in 3.7 the content-type
```
Comment (jsteemann) [inline arangod/RestHandler/RestReplicationHandler.cpp]:
```suggestion
  // simon: originally VST was not allowed here, but in 3.7 the content-type
```

File: arangod/GeneralServer/VstCommTask.cpp (hunk 2 above)
Comment (graetzer) [reply arangod/GeneralServer/VstCommTask.cpp]: no this is consistent with the per request logging in other protocol handlers

Comment (jsteemann) [reply arangod/GeneralServer/VstCommTask.cpp]: vale

File: arangod/Cluster/ClusterMethods.cpp (hunk 1 above)
Comment (graetzer) [reply arangod/Cluster/ClusterMethods.cpp]: good catch

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8481/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8487/

File: 3rdParty/fuerte/src/H2Connection.cpp [hunk 6]
```diff
@@ -846,14 +862,15 @@ bool H2Connection<T>::shouldStop() const {
 // ping ensures server does not close the connection
 template <SocketType T>
 void H2Connection<T>::startPing() {
+#if 0
```
Comment (jsteemann) [inline 3rdParty/fuerte/src/H2Connection.cpp]: `startPing()` is useless now? intentional?

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8491/

File: 3rdParty/fuerte/src/H2Connection.cpp (hunk 6 above)
Comment (graetzer) [reply 3rdParty/fuerte/src/H2Connection.cpp]: I will re-enable it later again

**Comment by graetzer** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8493/

**Pull request merged**